apps/astarte_housekeeping/lib/astarte_housekeeping/engine.ex

```diff
@@ -83,4 +83,83 @@ defmodule Astarte.Housekeeping.Engine do
   def list_realms do
     Queries.list_realms()
   end
+
+  @doc """
+  Updates a given realm using the provided values map. Fails if realm does not exists or values are invalid.
+  At the moment, only updating the realm jwt_public_key_pem is supported.
+  Returns a tuple:
+  - either {:ok, updated_realm}
+  - or {:error, reason} where reason is an atom describing the error.
+  """
+  @spec update_realm(binary(), %{:jwt_public_key_pem => binary(), optional(any()) => any()}) ::
```

Comment:
`realm_name` can probably be `String.t()` instead of `binary()` as we want it to be a printable string.
Same for `jwt_public_key_pem`.